app/services/oauth_clients/create_service.rb

```diff
@@ -36,10 +36,9 @@ module OAuthClients
   class CreateService < ::BaseServices::Create
     protected
 
-    def before_perform(params, _service_result)
-      OAuthClient.where(integration: params[:integration]).delete_all
-
-      super
+    def after_validate(params, contract_call)
+      OAuthClient.where(integration: params[:integration]).delete_all if contract_call.success?
```

Comment:
Hi @akabiru, Wieland and I QAed this PR earlier and found something small and not urgent here.

We realised that `if contract_call.success?` should not be needed as `after_validate` is only called when `contract_call.success?` [base_contracted.rb#L63](https://github.com/opf/openproject/blob/a06e519ed789984cdfb2ef80a1d760ffe23e3cb4/app/services/base_services/base_contracted.rb#L63)
Maybe there is anything I am missing, but if not it would be nice to remove this bit for clarity.

Reply:
That's a nice catch @dominic-braeunlein - I'll update this in dev 👍🏾

Reply:
Done! https://github.com/opf/openproject/pull/14449